playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2

```diff
@@ -22,8 +22,11 @@ http {
         # see http://orensol.com/2009/01/18/nginx-and-weird-400-bad-request-responses/
         large_client_header_buffers 4 16k;
         # server_tokens off;
-
+        {% if NGINX_OVERRIDE_SERVER_NAMES_HASH_BUCKET_SIZE %}
+        server_names_hash_bucket_size {{ NGINX_SERVER_NAMES_HASH_BUCKET_SIZE }};
+        {% else %}
         # server_names_hash_bucket_size 64;
```

Comment:
It looks like this is still commented out upstream: 

https://github.com/edx/configuration/blob/e02a6022b320087ca09211026ba55909eaa8ce68/playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2#L30

I'd recommend: 

- uncomment this line
- get rid of the `if NGINX_OVERRIDE_SERVER_NAMES_HASH_BUCKET_SIZE` conditional
- set a sane default for `NGINX_SERVER_NAMES_HASH_BUCKET_SIZE`
- push the change upstream

It looks like the default hash bucket size is 32 according to this doc: http://nginx.org/en/docs/http/server_names.html

That might be the appropriate value for upstream (and probably works well for edx.org subdomains), but is too small for most of our customers. We can override it in our own server-vars.yml and all of our base settings files.

Reply:
we did this change already on hawthorn (and ficus): https://github.com/appsembler/configuration/commit/61bffe30a9744377b46eecee09d3e118fbfb63d2

it just never made it into ginkgo.

we should also just increase the default from 64 to 128 while we are at it though.